common/lib/capa/capa/responsetypes.py

```diff
@@ -1099,22 +1228,91 @@ def setup_response(self):
     def get_score(self, student_answers):
         # log.debug('%s: student_answers=%s' % (unicode(self),student_answers))
         cmap = CorrectMap()
+        
         amap = self.get_answers()
+        
+        tree = self.xml
+        problem_xml = tree.xpath('.')
+
+        # Partial credit type - can set 'points' only at the moment.
+        credit_type = problem_xml[0].get('partial_credit', default=False)
+
+        try:
+            credit_type = str(credit_type).lower().strip()
+        except ValueError:
+            _ = self.capa_system.i18n.ugettext
+            # Translators: 'partial_credit' is an attribute name and should not be translated.
+            msg = _("partial_credit value can only be set to 'points' or be removed.")
+            raise LoncapaProblemError(msg)
+        
+
         for aid in amap:
-            if aid in student_answers and student_answers[aid] == amap[aid]:
-                cmap.set(aid, 'correct')
-            else:
-                cmap.set(aid, 'incorrect')
+            for word in amap[aid]:
```

Comment:
whoa, hahahaha, omg holy unreadable variable names Batman! Please add some comments explaining what's going on here. I know you didn't write all this code but obviously you've figured it out, please add in what `aid` and `amap` mean!~